### PR TITLE
introduced prow job to run KIND e2e tests for DWD

### PR DESCRIPTION
**Pull request by unmarshall**
**What this PR does / why we need it**:
DWD has e2e tests that bring up a vanilla KIND cluster via go code. To start a KIND cluster a different image and a privileged container is required. This PR adds a separate job only to run KIND e2e tests.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (oliver-goetz) [inline config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml:17]: please remove `fork-per-release` annotation. The `job-forker` is enabled for g/g only atm.

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (oliver-goetz) [inline config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml:57]: same here with `fork-per-release`

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (oliver-goetz) [inline config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml:37]: Are you using `skaffold` for your tests? If not you could remove these envs.

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (oliver-goetz) [inline config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml:77]: same here with `skaffold`

**Review by oliver-goetz — COMMENTED** [review]
Found some small things.
Are you aware that the jobs will fail until you merged the refactoring PR because the make target is not available on master yet. Until then, the only place where it is able to run will be the refactoring PR itself.

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: done

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: done

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: removed as we are not using skaffold

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: removed as we are not using skaffold

**Comment by unmarshall** [comment]
Thanks for the quick review. I have addressed the review comments. PTAL.

**Comment by oliver-goetz** [comment]
/lgtm
/approve
/hold
@unmarshall please `/unhold` the PR when you are ready to go

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (timebertt) [inline config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml:3]: "kind unit tests" sounds like a contradiction in itself to me.
I expect unit tests not to have any external dependencies like a running k8s cluster (ref [testing guidelines](https://github.com/gardener/gardener/blob/master/docs/development/testing.md)).

Are these really "unit tests"? If not, please align the job name with our testing guidelines.
If they are "unit tests", then please reconsider the testing architecture in DWD.

**Review by timebertt — CHANGES_REQUESTED** [review]

File: config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml
Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml:3]: > "kind unit tests" sounds like a contradiction in itself to me.

I do not necessarily agree to this but i will make the change to follow the naming convention. We do only start a vanilla KIND cluster and do not need any gardener components.

Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: Changes are pushed. PTAL.

Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: DWD mainly requires a functional KAPI, etcd and KCM. For rest of the deployments like MCM and CA we start dummy deployments and change their status and replicas dynamically. So we start a KIND cluster to test the e2e functionality of prober. For weeder (which was previously called endpoint) we do not need any KIND tests.

Comment (timebertt) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: Do you really need KCM? If not, you might be interested in envtest, which is way easier to execute locally.

Comment (rfranzke) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: Not sure about the details, but if it's just about updating some status that normally KCM would do, you can also fake its behaviour in your test (by manually updating it).

Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: Ok if there are many concerns then we will continue using concourse pipeline for KIND cluster tests. For release we will anyways use concourse pipeline. I can then close this PR. Let me know.

Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: I would request that you allow us to create this job as these are e2e tests but without gardener setup. We will rework these tests and perhaps re-use what g/g provides so that we test against gardener KIND cluster. Similar to what what is done in [extension-shoot-oidc-service](https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/master/hack/test-e2e-provider-local.sh)

Comment (rfranzke) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: I think nobody wanted to deny/block this PR. There was just some confusion about the naming which triggered the suggestions above. Please go ahead. cc @oliver-goetz

Comment (unmarshall) [reply config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml]: Thanks @rfranzke 
I have removed the hold.

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**
